core/src/main/java/net/neoforged/fml/config/ConfigFileTypeHandler.java

```diff
@@ -31,38 +31,56 @@ public class ConfigFileTypeHandler {
     public Function<ModConfig, CommentedFileConfig> reader(Path configBasePath) {
         return (c) -> {
             final Path configPath = configBasePath.resolve(c.getFileName());
-            final CommentedFileConfig configData = CommentedFileConfig.builder(configPath).sync().
-                    preserveInsertionOrder().
-                    autosave().
-                    onFileNotFound((newfile, configFormat)-> setupConfigFile(c, newfile, configFormat)).
-                    writingMode(WritingMode.REPLACE).
-                    build();
-            LOGGER.debug(CONFIG, "Built TOML config for {}", configPath.toString());
+            final CommentedFileConfig configData = CommentedFileConfig.builder(configPath)
+                    .sync()
+                    .preserveInsertionOrder()
+                    .autosave()
+                    .onFileNotFound((newfile, configFormat) -> setupConfigFile(c, newfile, configFormat))
+                    .writingMode(WritingMode.REPLACE)
+                    .build();
+            LOGGER.debug(CONFIG, "Built TOML config for {}", configPath);
             try
             {
                 configData.load();
             }
             catch (ParsingException ex)
             {
-                throw new ConfigLoadingException(c, ex);
+                LOGGER.warn(CONFIG, "Attempting to recreate {}", configPath);
+                try
+                {
+                    backUpConfig(configData.getNioPath(), FMLConfig.getIntConfigValue(FMLConfig.ConfigValue.CONFIG_BACKUPS_COUNT));
+                    Files.delete(configData.getNioPath());
+
+                    configData.load();
+                }
+                catch (Throwable t)
+                {
+                    ex.addSuppressed(t);
+
+                    throw new ConfigLoadingException(c, ex);
+                }
```

Comment:
What is this for?

Reply:
For something like this https://github.com/mezz/JustEnoughItems/issues/3317